teamsupport/models.py

```diff
@@ -0,0 +1,66 @@
+from property_caching import cached_property
+
+
+class Ticket(object):
+    _data = None
+
+    def __init__(self, client, ticket_id=None, data=None):
+        self.client = client
+        if data is not None:
+            self._data = data
+            self.id = data.find('TicketID').text
+        elif ticket_id:
+            self.id = ticket_id
+        elif not (ticket_id or data):
+            raise TypeError(
+                "__init__() needs either a 'ticket_id' or 'data' argument "
+                "(neither given)")
+
+    def __getattr__(self, name):
+        if self.data.find(name) is not None:
+            return self.data.find(name)
+        raise AttributeError(name)
+
+    @property
+    def data(self):
+        if self._data is None:
+            self._data = self.client.get_ticket(self.id)
```

Comment:
Are we gaining much by this indirection? We could call `get_ticket` in `__init__` and things would be a bit simpler.